exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,618 @@
+#pragma once
+
+#include <sparkplug_b/sparkplug_b.pb.h>
+#include <async_mqtt/all.hpp>
+#include <boost/asio.hpp>
+
+#include <tfc/dbus/string_maker.hpp>
+
+#include "config.hpp"
+
+using org::eclipse::tahu::protobuf::Payload;
+using org::eclipse::tahu::protobuf::Payload_Metric;
+
+namespace asio = boost::asio;
+namespace details = tfc::ipc::details;
+
+class network_manager {
+public:
+  auto connect_socket(auto&& socket, auto&& resolved_ip) -> asio::awaitable<void> {
+    co_await asio::async_connect(std::forward<decltype(socket)>(socket), std::forward<decltype(resolved_ip)>(resolved_ip),
+                                 asio::use_awaitable);
+  }
+
+  auto set_sni_hostname(auto&& native_handle, std::string const& broker_address) -> void {
+    if (!SSL_set_tlsext_host_name(std::forward<decltype(native_handle)>(native_handle), broker_address.c_str())) {
+      const boost::system::error_code error{ static_cast<int>(::ERR_get_error()), boost::asio::error::get_ssl_category() };
+      throw boost::system::system_error{ error };
+    }
+  }
+
+  auto async_handshake(auto&& socket) -> asio::awaitable<void> {
+    co_await std::forward<decltype(socket)>(socket).async_handshake(async_mqtt::tls::stream_base::client,
+                                                                    asio::use_awaitable);
+  }
+};
```

Comment:
I am not really a fan of this but don't have an easy solution that
works instead.

Reply:
agree